data/data/openstack/masters/outputs.tf

```
@@ -0,0 +1,13 @@
+output "control_plane_ips" {
+  value = <<EOT
+    var.master_count > 2 ? [
+      openstack_compute_instance_v2.master_conf_0.access_ip_v4,
+      openstack_compute_instance_v2.master_conf_1.access_ip_v4,
+      openstack_compute_instance_v2.master_conf_2.access_ip_v4 ] :
+        var.master_count > 1 ? [
+          openstack_compute_instance_v2.master_conf_0.access_ip_v4,
+          openstack_compute_instance_v2.master_conf_1.access_ip_v4 ] :
+            var.master_count > 0 ? [
+              openstack_compute_instance_v2.master_conf_0.access_ip_v4 ] : 0
+EOT
```

This does not look right to me. Each of the `openstack_compute_instance_v2.master_conf_?` are a collection of resources. They require a selection to be made.
```suggestion
  value = concat(
      openstack_compute_instance_v2.master_conf_0.*.access_ip_v4,
      openstack_compute_instance_v2.master_conf_1.*.access_ip_v4,
      openstack_compute_instance_v2.master_conf_2.*.access_ip_v4,
    )
```

---

Actually, what you have ends up outputting the text in between the EOT markers rather than a list of IPs.
```
time="2021-09-27T17:45:27Z" level=debug msg="control_plane_ips =     var.master_count > 2 ? ["
time="2021-09-27T17:45:27Z" level=debug msg="      openstack_compute_instance_v2.master_conf_0.access_ip_v4,"
time="2021-09-27T17:45:27Z" level=debug msg="      openstack_compute_instance_v2.master_conf_1.access_ip_v4,"
time="2021-09-27T17:45:27Z" level=debug msg="      openstack_compute_instance_v2.master_conf_2.access_ip_v4 ] :"
time="2021-09-27T17:45:27Z" level=debug msg="        var.master_count > 1 ? ["
time="2021-09-27T17:45:27Z" level=debug msg="          openstack_compute_instance_v2.master_conf_0.access_ip_v4,"
time="2021-09-27T17:45:27Z" level=debug msg="          openstack_compute_instance_v2.master_conf_1.access_ip_v4 ] :"
time="2021-09-27T17:45:27Z" level=debug msg="            var.master_count > 0 ? ["
time="2021-09-27T17:45:27Z" level=debug msg="              openstack_compute_instance_v2.master_conf_0.access_ip_v4 ] : 0"
```